dashboard/test/controllers/levels_controller_test.rb

```diff
@@ -72,6 +72,55 @@ def create_level_with_rubric
     )
   end
 
+  test "should get filters" do
+    get :get_filters, params: {}
+    assert_equal JSON.parse(@response.body)["levelOptions"], [['All types', ''], *LevelsController::LEVEL_CLASSES.map {|x| [x.name, x.name]}.sort_by {|a| a[0]}]
+    assert_equal JSON.parse(@response.body)["scriptOptions"], [
+      ['All scripts', ''],
+      *Script.valid_scripts(@levelbuilder).pluck(:name, :id).sort_by {|a| a[0]}
+    ]
+    assert_equal JSON.parse(@response.body)["ownerOptions"], [
+      ['Any owner', ''],
+      *Level.joins(:user).distinct.pluck('users.name, users.id').sort_by {|a| a[0]}
+    ]
+  end
+
+  test "should get filtered levels with just page param" do
+    get :get_filtered_levels, params: {page: 1}
+    assert_equal JSON.parse(@response.body).length, 30
```

Comment:
I think checking for the exact levels won't work over time as new levels are add so I just went with checking count

Reply:
I think only the data in fixtures should affect unit tests, right?

Reply:
I think only the levels added through fixtures should affect the unit tests, right?

Reply:
Ya I'm worried about the names of the tests that get set up by factories since they are based on number and when levels are created. I'm giving it a shot tho

Reply:
Ya the drone run failed: 

============== FAIL["test_should_get_filtered_levels_with_just_page_param", "LevelsControllerTest", 973.1403690139996]
--
4582 | test_should_get_filtered_levels_with_just_page_param#LevelsControllerTest (973.14s)
4583 | --- expected
4584 | +++ actual
4585 | @@ -1 +1 @@
4586 | -["Level_1350", "Level_1351", "Level_122", "Level_123", "Level_124", "Level_125", "ScriptActivity test level 1", "ScriptActivity test level 2", "Test longassessment levelgroup split new", "Sample CSP Assessment", "Test longassessment levelgroup new", "Test anonymous student survey", "PS U3 Assessment1 new", "Pluralsight LevelGroup Test", "CSP pre-assessment survey", "U3L25 Student Lesson Introduction", "U3L20 Student Lesson Introduction", "blockly", "blockly", "blockly", "blockly", "allthethings_fish_tank", "courseC_PlayLab_events1", "iceage_warn_ice_age", "iceage_move_events", "iceage_hello2", "iceage_move_to_flag", "iceage_sound_and_points", "iceage_move_to_actor", "iceage_free_play"]
4587 | +["Level_81", "Level_82", "Test longassessment levelgroup split new", "Sample CSP Assessment", "Test longassessment levelgroup new", "Test anonymous student survey", "PS U3 Assessment1 new", "Pluralsight LevelGroup Test", "CSP pre-assessment survey", "U3L25 Student Lesson Introduction", "U3L20 Student Lesson Introduction", "blockly", "blockly", "blockly", "blockly", "allthethings_fish_tank", "courseC_PlayLab_events1", "iceage_warn_ice_age", "iceage_move_events", "iceage_hello2", "iceage_move_to_flag", "iceage_sound_and_points", "iceage_move_to_actor", "iceage_free_play", "iceage_throw_hearts", "scrat 17", "scrat 19", "scrat 18", "scrat 20", "scrat 16"]

Reply:
Also similar problem here: 

= FAIL["test_should_get_filters", "LevelsControllerTest", 974.0702120909991]
--
4592 | test_should_get_filters#LevelsControllerTest (974.07s)
4593 | --- expected
4594 | +++ actual
4595 | @@ -1 +1 @@
4596 | -["All scripts", "20-hour", "algebra", "artist", "bogus-script-150", "bogus-script-183", "bogus-script-184", "bogus-script-185", "course1", "course2", "course3", "course4", "coursea-2017", "courseb-2017", "coursec-2017", "coursed-2017", "coursee-2017", "coursef-2017", "express-2017", "flappy", "frozen", "hourofcode", "jigsaw", "playlab", "pre-express-2017", "starwars"]
4597 | +["All scripts", "20-hour", "algebra", "artist", "course1", "course2", "course3", "course4", "coursea-2017", "courseb-2017", "coursec-2017", "coursed-2017", "coursee-2017", "coursef-2017", "express-2017", "flappy", "frozen", "hourofcode", "jigsaw", "playlab", "pre-express-2017", "starwars"]

Reply:
Thoughts on best way to deal with this @davidsbailey ?

Reply:
it seems like there are a number of factors which could affect which levels are in the DB before your test runs, including:
* anything in your local test DB added over time
* new fixtures added over time
* any unit tests which fail to clean up after themselves

so, I would recommend against writing a test which assumes a hard-coded list of levels already in the DB.

A few ideas for how to work around this:
1. always filter on something (script or owner) so that you can be sure to exclude any content not generated by your current test
2. trusting that your test case is wrapped in a transaction, delete all levels, then add back only the ones you want to test against. (could be slow. might be best to do as part of an "integration" test. only do this in one test case to minimize the performance hit.)

Reply:
Are the other tests I have enough? I have tests with each of the filters using a specific value

Reply:
yes, in combination with your other tests, just testing the count here is probably enough.